### ISSUE TITLE
Using TochMetrics for code simplification

### ISSUE THREAD

**Issue by Borda** [ppogg/YOLOv5-Lite#147]
# :hammer_and_wrench: Proposed Refactor
We have been developing [TochMetrics](https://github.com/PyTorchLightning/metrics) to be a general-purpose metric and some domain-specific use-cases. In many cases, we have an exact mapping to scikit-learn with verification/testing to the reference metric for its correctness. The TM includes functional as well as `nn.Module` versions and for the most/standard metrics, the only dependency is PyTorch (for the domain-specific metrics you need to install related extras).

## Suggest a potential alternative/fix

WIth using [TM metrics](https://torchmetrics.readthedocs.io/en/stable/) you may rely on the widely tested correctness (testing against good standard `pycocotools` in multiple OS environments and all PyTorch versions above v1.4) and later you can use `nn.Module` to leverage update and compute.

Overall if you are fine with it we are happy to draft a PR with a suggested change to verify in place the impact.
If you have any questions, happy to follow up with me or @aniketmaurya

What I have quickly checked, all this [utils.metrics](https://github.com/ppogg/YOLOv5-Lite/blob/master/utils/metrics.py) can be replaced by TM

**Comment by ppogg** [comment]
Wow! It's a really useful tool !
